test/unit/configuration_test.rb

```diff
@@ -26,31 +36,31 @@ class ConfigurationTest < Minitest::Test
         "custom_associations" => ["custom_association"],
         "inflections_file" => "custom_inflections.yml",
       }
-      YAML.expects(:load_file).with("./packwerk.yml").returns(configuration_hash)
+      merge_into_yaml_file("packwerk.yml", configuration_hash)
```

Comment:
Hmmm, reading the test case I find the naming of the helper methods a bit confusing. Maybe we should follow a naming standard so it is clear that we are working with the fixture helpers. Maybe `app_fixture_*` -> 
```ruby
app_fixture_use_template(:minimal)
app_fixture_merge_yaml_file(...) # or app_fixture_override_package_config(...)
app_fixture_remove_file("packwerk.yml") # instead of remove_app_entry()
```


What do you think?

Reply:
I don't think the suggested prefix adds much clarity, nor does it follow a common naming pattern.  If someone is confused, they'll find the helper module soon enough.

I find that `merge_into_yaml_file`, although not perfect, is truer to the actual behaviour than `override_package_config`.

I went with `remove_app_entry` since it models https://ruby-doc.org/stdlib-2.7.0/libdoc/fileutils/rdoc/FileUtils.html#method-c-remove_entry in its behaviour (works on directories as well, is idempotent, works on missing entries).

I will go with what I have for now and we can revisit later.

Reply:
My issue with the current names is that they suggest behavior that they don't have.
For example, if I read `merge_into_yaml_file`, I would think of a method that takes a path to a YAML file and merges a hash with it. That is not what it does, though. It assumes a path for the fixture directory and merges a hash with a file in that directory.

Thinking about it, it seems like we have two functionalities. One is for setting up a fixture helper, and another for methods for manipulating IO. It would become clearer if we separate the two responsibilities, and the current names would make more sense. So, for example:

```ruby 
module FileHelper
  # If you look at those two methods, you can see that they are quite separate from the fixture functionality.
  def merge_into_yaml_file(path, hash)
    YamlFile.new(path).merge(hash)
  end

  def remove_file(path)
    FileUtils.remove_entry(path)
  end
```

And then we can call it like this:

```ruby
remove_app_entry(path_to("packwerk.yml"))
merge_into_yaml_file(path_to("packwerk.yml"), configuration_hash)
```

Reply:
If we separate the responsibilities, we can also make the other names a bit clearer. 

I would suggest to rename `copy_template(:minimal)` to `use_template(:minimal)`. Copy, in this case, describes how the fixture helper is setting up the template. Since a user can effectively use only one template, I would go with `use_template(:minimal)` here. It abstracts how the feature is implemented from the intended outcome. 

We could also clear up `path_to` because it is not clear where we are creating the path. I want to avoid someone needing to hunt down the implementation to understand what is happening in a test case. What do you think of something like `expand_app_fixture_path`, `expand_app_path` or `to_app_path`?

Reply:
You listed a lot of good alternatives to what I currently have.  Comparing this first approach
```ruby
remove_entry(path_to("packwerk.yml")) 
merge_into_yaml_file(path_to("components/sales/package.yml"), { enforce_privacy: false })
```
To this second one
```ruby
remove_app_entry("packwerk.yml")
merge_into_app_yaml_file("components/sales/package.yml", { enforce_privacy: false })
```
I like the second one better.  Adding `app` everywhere should clear things up.

I will rename `path_to` as well
```ruby
assert_equal to_app_paths(["components/sales"]), config.load_paths
```